src/Aspire.Dashboard/Components/Controls/Grid/AspirePropertyColumn.cs

```diff
@@ -1,14 +1,39 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components;
 using Microsoft.FluentUI.AspNetCore.Components;
 
 namespace Aspire.Dashboard.Components.Controls.Grid;
 
-public class AspirePropertyColumn<TGridItem, TProp> : PropertyColumn<TGridItem, TProp>
+public class AspirePropertyColumn<TGridItem, TProp> : PropertyColumn<TGridItem, TProp>, IAspireColumn
 {
+    [Parameter]
+    public GridColumnManager? ColumnManager { get; set; }
+
+    [Parameter]
+    public string? ColumnId { get; set; }
+
+    [Parameter]
+    public bool UseCustomHeaderTemplate { get; set; } = true;
+
     protected override void OnParametersSet()
     {
-        HeaderCellItemTemplate = AspireFluentDataGridHeaderCell.RenderHeaderContent(Grid);
         base.OnParametersSet();
+
+        if (UseCustomHeaderTemplate)
+        {
+            HeaderCellItemTemplate = AspireFluentDataGridHeaderCell.RenderHeaderContent(Grid);
+        }
+    }
+
+    protected override bool ShouldRender()
+    {
+        if (ColumnManager is not null && ColumnId is not null && !ColumnManager.IsColumnVisible(ColumnId))
+        {
+            return false;
+        }
+
+        return base.ShouldRender();
     }
```

Comment:
Nice. I was thinking about something like this. `@if (xxx) { ... }` seemed to be the idiomatic way to conditionally show UI in Razor, but building the condition into the column makes it much cleaner.